### PR TITLE
Add mode validation

### PR DESCRIPTION
**Pull request by zptro**
Add feature to netowork validation: Check that the modes defined in assignment parameters are actually found in the network.

To be able to test this feature, mock assignment module needed some extensions.

### CODE REVIEW

**Comment by johpiip** [comment]
Can you explain the purpose of this PR, thanks! :)

**Comment by zptro** [comment]
> Can you explain the purpose of this PR, thanks! :)

Sorry, now I have added a short description.

File: Scripts/utils/validate_network.py [hunk 1]
```diff
@@ -42,6 +42,13 @@ def validate(network, fares=None):
                 found_zone_share)
             log.error(msg)
             raise ValueError(msg)
+    validate_mode(network, param.main_mode, "AUTO")
```
Comment (johpiip) [inline Scripts/utils/validate_network.py:45]: Are "AUTO", "AUX_AUTO", "TRANSIT", and "AUX_TRANSIT" here names that come from how Emme API reads networks? Consider moving these magic variables as a parameter.

**Review by johpiip — COMMENTED** [review]
I can't do any tests right now because I have model runs going on *both* of our Emme machines, so I will continue next week. 😄

File: Scripts/utils/validate_network.py (hunk 1 above)
Comment (zptro) [reply Scripts/utils/validate_network.py]: Yes, they come from the Emme text file import syntax. Now I moved the strings to the beginning of the file as constants.

**Review by johpiip — APPROVED** [review]
Awesome, thanks!

**Pull request merged**